src/sagas/ui.js

```diff
@@ -29,9 +33,30 @@ export function* exportGist() {
   }
 }
 
+export function* exportRepo() {
+  const exportWindow = yield call(
+    openWindowWithWorkaroundForChromeClosingBug,
+    `data:text/html;base64,${spinnerPage}`,
+  );
+  const {type, payload: url} =
+    yield take(['REPO_EXPORTED', 'REPO_EXPORT_ERROR']);
+
+  if (type === 'REPO_EXPORTED') {
+    if (exportWindow.closed) {
+      yield put(repoExportNotDisplayed(url));
+    } else {
+      exportWindow.location.href = url;
+      yield put(repoExportDisplayed());
+    }
+  } else {
+    yield call([exportWindow, 'close']);
+  }
+}
```

Comment:
This looks very similar to `exportGist`—can we extract a generic function that both sagas can call?